### PR TITLE
ci: build with crash reporter, auto updater and appimage updater

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
that did not work .....

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -62,116 +62,120 @@ jobs:
...
+        run: |
+          if ("${{ matrix.target }}" -eq "macos-64-clang-debug" ) {
+              # https://api.kde.org/ecm/module/ECMEnableSanitizers.html
+              # address;leak;undefined
+              # clang: error: unsupported option '-fsanitize=leak' for target 'x86_64-apple-darwin21.6.0'
+              & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c --set args="-DECM_ENABLE_SANITIZERS='address;undefined'" owncloud/owncloud-client
+          }
+          if ("${{ matrix.target }}" -eq "linux-gcc-x86_64" ) {
+              # build with appimage updater and others
+              & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c --set args="-DWITH_CRASHREPORTER=ON -DWITH_AUTO_UPDATER=ON -DWITH_APPIMAGEUPDATER=ON" owncloud/owncloud-client
```
Comment (erikjv) [inline .github/workflows/main.yml:141]: see above

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -62,116 +62,120 @@ jobs:
...
+          & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c --set srcDir="${env:GITHUB_WORKSPACE}" owncloud/owncloud-client
+          if ($IsWindows) {
+              & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c dev-utils/nsis
+          } elseif($IsLinux) {
+              & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c dev-utils/linuxdeploy
+          }
+
+      - name: Install dependencies
+        run: |
+          & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c --install-deps owncloud/owncloud-client
```
Comment (erikjv) [inline .github/workflows/main.yml:106]: On Linux I run:

```
craft --set enableAppImageUpdater=True owncloud-client
```

before installing the deps. Then CMake will pick up the package, and turn that feature on without needing extra `-D` flags.

**Review by erikjv — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 2 above)
Comment (DeepDiver1975) [reply .github/workflows/main.yml:106]: does something similar exist for crash reporter and auto updater? @erikjv

Comment (DeepDiver1975) [reply .github/workflows/main.yml:110]: that is also not working - no trace of appimageupdater.cpp being compiled ....

**Comment by DeepDiver1975** [comment]
:tada:

**Comment by DeepDiver1975** [comment]
@erikjv using --set sent me on the right track - thx

File: .github/workflows/main.yml (hunk 2 above)
Comment (erikjv) [reply .github/workflows/main.yml:112]: > does something similar exist for crash reporter and auto updater? @erikjv

`craft --search owncloud-client` gives all the options

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -62,116 +62,128 @@ jobs:
...
+        run: |
+          if ("${{ matrix.target }}" -eq "macos-64-clang-debug" ) {
+              # https://api.kde.org/ecm/module/ECMEnableSanitizers.html
+              # address;leak;undefined
+              # clang: error: unsupported option '-fsanitize=leak' for target 'x86_64-apple-darwin21.6.0'
+              & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c --set args="-DECM_ENABLE_SANITIZERS='address;undefined'" owncloud/owncloud-client
+          }
+          if ("${{ matrix.target }}" -eq "linux-64-gcc" ) {
+              # build with appimage updater and others
+              & "${env:GITHUB_WORKSPACE}/.github/workflows/.craft.ps1" -c --set enableAppImageUpdater=true owncloud-client
```
Comment (erikjv) [inline .github/workflows/main.yml:147]: Do you need to do this again? This is already done in the `Install dependencies` step.

Comment (DeepDiver1975) [reply .github/workflows/main.yml:147]: it has to be done again to have the env properly setup - at least this is the way I understood this.

**Pull request closed without merging**